haystack/components/joiners/document_joiner.py

```diff
@@ -197,7 +197,10 @@ def _distribution_based_rank_fusion(self, document_lists: List[List[Document]])
             max_score = mean_score + 3 * std_dev
 
             for doc in documents:
-                doc.score = (doc.score - min_score) / (max_score - min_score)
+                try:
+                    doc.score = (doc.score - min_score) / (max_score - min_score)
+                except ZeroDivisionError:
+                    doc.score = 0  # if max and min is the same value all documents are poorly informative
```

Comment:
We don't need exception handling here - Let's save `max - min` outside of the loop, check the value and set the score to zero if necessary.